scIB/metrics.py

```diff
@@ -1728,28 +1741,40 @@ def get_root(adata_pre, adata_post, ct_key, dpt_dim=3):
             opt = np.argmin
         else:
             opt = np.argmax
+
         # count opt cell
         if len(diffmap_min_dpt) == 0:
             raise RootCellError('No root cell in largest component')
+
         min_dpt_cell[opt(diffmap_min_dpt)] += 1
     
     # root cell is cell with max vote
     return min_dpt[np.argmax(min_dpt_cell)], adata_post_sub
 
-def trajectory_conservation(adata_pre, adata_post, label_key):
-    cell_subset = adata_pre.obs.index[adata_pre.obs["dpt_pseudotime"].notnull()]
+def trajectory_conservation(adata_pre, adata_post, label_key, pseudotime_key="dpt_pseudotime"):
+    """
+    :param adata_pre: unintegrated adata
+    :param adata_post: integrated adata
+    :param label_key: column in `adata_pre.obs` of the groups used to precompute the trajectory
+    :param pseudotime_key: column in `adata_pre.obs` in which the pseudotime is saved in.
+        Column can contain empty entries, the dataset will be subset to the cells with scores.
+    """
+    # subset to cells for which pseudotime has been computed
+    cell_subset = adata_pre.obs.index[adata_pre.obs[pseudotime_key].notnull()]
     adata_pre_sub = adata_pre[cell_subset]
     adata_post_sub = adata_post[cell_subset]
-    
-    iroot, adata_post_sub2 = get_root(adata_pre_sub, adata_post_sub, label_key)
+
+    iroot, adata_post_sub2 = get_root(adata_pre_sub, adata_post_sub, label_key, pseudotime_key)
     adata_post_sub2.uns['iroot'] = iroot
     
-    sc.tl.dpt(adata_post_sub2)
-    adata_post_sub2.obs['dpt_pseudotime'][adata_post_sub2.obs['dpt_pseudotime']>1]=0
+    sc.tl.dpt(adata_post_sub2)  # stored in 'dpt_pseudotime'
+    adata_post_sub2.obs['dpt_pseudotime'][adata_post_sub2.obs['dpt_pseudotime'] > 1] = 0
     adata_post_sub.obs['dpt_pseudotime'] = 0
     adata_post_sub.obs['dpt_pseudotime'] = adata_post_sub2.obs['dpt_pseudotime']
-    adata_post_sub.obs['dpt_pseudotime'].fillna(0)
```

Comment:
removing this line means you have na in the entries that were 0 before, no? That would change the metric.